harness/determined/pytorch/_lightning/adapter.py

```diff
@@ -0,0 +1,275 @@
+        self.lm = lm
+        self.optimizers = optimizers
+
+
+class PLAdapter(PyTorchTrial):
+    def __init__(self, context: PyTorchTrialContext, lightning_module: pl.LightningModule):
+        super().__init__(context)
+        check_compatibility(lightning_module)
+        override_unsupported_nud(lightning_module, context)
+        context.wrap_model(lightning_module)
+        optimizers, lr_schedulers = self.setup_optimizers_schedulers(context, lightning_module)
+        pls = _PLAdapterState(context, lightning_module, optimizers)
+        self._pls = pls
+
+        # set lightning_module properties
+        pls.lm.use_ddp = False  # type: ignore
+        pls.lm.use_ddp2 = False  # type: ignore
+        pls.lm.use_dp = False  # type: ignore
+        pls.lm.use_tpu = False  # type: ignore
+        type(pls.lm).local_rank = context.distributed.get_local_rank()  # type: ignore
+        type(pls.lm).global_rank = context.distributed.get_rank()  # type: ignore
+        pls.lm.use_amp = context.experimental._auto_amp or context._use_apex
+        pls.lm.to(context.device)
+
+    def build_callbacks(self) -> Dict[str, PyTorchCallback]:
+        """
+        build_callbacks defines a set of necessary PyTorchTrialCallback to support
+        lightning. Override and merge the output of this build_callbacks with your
+        desired callbacks.
+        """
+        context = self._pls.context
+        lm = self._pls.lm
+
+        class PLAdapterCallback(PyTorchCallback):
+            def on_training_epoch_start(self) -> None:
+                if context._current_batch_idx is not None:
+                    type(lm).current_epoch = context.current_train_epoch()  # type: ignore
+                lm.on_train_epoch_start()
+
+            def on_validation_epoch_start(self) -> None:
+                lm.on_validation_epoch_start()
+
+            def on_validation_epoch_end(self, outputs: List[Any]) -> None:
+                lm.on_validation_epoch_end()
+                lm.validation_epoch_end(outputs)
+
+        return {"_lightning_module": PLAdapterCallback()}
+
+    def setup_optimizers_schedulers(
+        self,
+        context: PyTorchTrialContext,
+        lightning_module: pl.LightningModule,
+    ) -> Tuple[List[Optimizer], List[_LRScheduler]]:
+        """
+        Wrap optimizers and lr_schedulers returned by `configure_optimizers` to
+        work with Determined.
+        Return: Wrapped `optimizers`, and `lr_schedulers` in a tuple
+        """
+        optimizers, lr_scheduler_dicts, opt_frequencies = TrainerOptimizersMixin().init_optimizers(
+            lightning_module,
+        )
+        for freq in opt_frequencies:
+            check.eq(freq, 1, "custom optimizer frequencies are not supported")
+        optimizers = cast(List[Optimizer], optimizers)
+        lr_scheduler_dicts = cast(List[dict], lr_scheduler_dicts)
+
+        def lightning_scheduler_dict_to_det(lrs: dict) -> _LRScheduler:
+            """
+            input_dict = {
```

Comment:
blocking: Need to add checks for the fields that we don't support. We need to support frequency. If `monitor` and `strict` are set, we should add checks and fail.

Reply:
> We need to support frequency.

let's leave this for a follow-up pr. I'll add the checks

Reply:
https://determinedai.atlassian.net/browse/DET-5021